packages/sdk/src/conversation/middlewares/commandMiddleware.ts

```diff
@@ -62,36 +62,39 @@ export class CommandResponseMiddleware implements Middleware {
       ssoHandler.triggerPatterns
     );
     this.ssoCommandHandlers.push(ssoHandler);
-    this.commandHandlers.push(ssoHandler);
     this.hasSsoCommand = true;
   }
 
   public async onTurn(context: TurnContext, next: () => Promise<void>): Promise<void> {
     if (context.activity.type === ActivityTypes.Message) {
       // Invoke corresponding command handler for the command response
       const commandText = this.getActivityText(context.activity);
-
+      let alreadyProcessed = false;
       for (const handler of this.commandHandlers) {
         const matchResult = this.shouldTrigger(handler.triggerPatterns, commandText);
 
         // It is important to note that the command bot will stop processing handlers
         // when the first command handler is matched.
         if (!!matchResult) {
-          if (this.isSsoExecutionHandler(handler)) {
+          const message: CommandMessage = {
+            text: commandText,
+          };
+          message.matches = Array.isArray(matchResult) ? matchResult : void 0;
+          const response = await handler.handleCommandReceived(context, message);
+
+          await this.processResponse(context, response);
+          alreadyProcessed = true;
+          break;
+        }
+      }
+
+      if (!alreadyProcessed) {
+        for (const handler of this.ssoCommandHandlers) {
+          const matchResult = this.shouldTrigger(handler.triggerPatterns, commandText);
+          if (!!matchResult) {
             await this.ssoActivityHandler?.run(context);
-          } else {
-            const message: CommandMessage = {
-              text: commandText,
-            };
-            message.matches = Array.isArray(matchResult) ? matchResult : void 0;
-            const response = await (handler as TeamsFxBotCommandHandler).handleCommandReceived(
-              context,
-              message
-            );
-
-            await this.processResponse(context, response);
+            break;
           }
-          break;
         }
       }
     } else {
```

Comment:
Do we need the `else` code block here if the command is responded in a non-sso command handler?

Reply:
This `else` condition is for none message activity type, such as `signin/tokenExchange` which will trigger login process

Reply:
I think we do not need to process the command twice, if non-sso command has already triggered, then it will break and stop trigger

Reply:
Oh ~ I see, I mistook the line indents, sorry for that.